File: Java/백준/Bronze/6378. 디지털 루트/디지털 루트.java
import java.util.Scanner;

public class Main{
    public static int dsum(String a){
        int sum = 0;
        for(int i=0; i<a.length(); i++){
            sum += Integer.parseInt(Character.toString(a.charAt(i)));
        }
        if(sum>=10){
            return dsum(Integer.toString(sum));
        }
        else{
            return sum;
        }
    }
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        while(true){
            String a = in.next();
            if(a.charAt(0)=='0'){
                break;
            }
            System.out.println(dsum(a));
        }
    }
}

File: C#/백준/Bronze/6378. 디지털 루트/디지털 루트.cs
using System;

namespace q{
    class w{
        static int dsum(string a){
            int sum = 0;
            for (int i=0; i<a.Length; i++){
                sum+=int.Parse(a[i].ToString());
            }
            if(sum>=10){
                return dsum(sum.ToString());
            }
            else{
                return sum;
            }
        }
        static void Main(){
            while(true){
                string a = Console.ReadLine();
                if(a[0]=='0'){
                    break;
                }
                Console.WriteLine(dsum(a));
            }
        }
    }
}